### DISCUSSION TITLE
Better default names for Filesystem Music Providers, easier custom name setup

### DISCUSSION THREAD

**Discussion by busywait** [music-assistant/support#2179] [Feature requests and ideas]
When I add several directories as Music Providers they all appear with the name  Filesystem (local disk), and I can't tell which folder is which. So, I have to choose "Configure" and add a custom name. 

It would be helpful if
1/ the provider name defaulted to or included the directory name, or just the last part of the full directory path
2/ the initial setup screen allowed the default custom name to be edited 

![image](https://github.com/music-assistant/support/assets/3671165/5310412d-6378-4c65-a8e9-cc5b7696fab4)

Currently the only way to add a custom name is to first add the provider and then edit it.

![image](https://github.com/music-assistant/support/assets/3671165/3956473f-0c88-4ed3-a84d-694739d88e7e)

**Comment by OzGav** [comment]
This has been implemented in MA 2.4

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
